src/main/cpp/syslogappender.cpp

```diff
@@ -326,11 +327,17 @@ void SyslogAppender::append(const spi::LoggingEventPtr& event, Pool& p)
 
 		while ( start != msg.end() )
 		{
-			LogString::iterator end = start + maxMessageLength - 12;
+			LogString::iterator end;// = start + maxMessageLength - 12;
+
+//			if ( end > msg.end() )
+//			{
+//				end = msg.end();
+//			}
 
-			if ( end > msg.end() )
-			{
+			if( msg.size() > (maxMessageLength - MINIMUM_MESSAGE_SIZE) ) {
```

Comment:
`maxMessageLength - MINIMUM_MESSAGE_SIZE` looks like some sort of `msgWindowLen[gth]`? Doesn't seem to change, so might be maintained on instance level, calculated with `setMaxMessageLength`. Makes code easier to understand in my opinion, like with removing `12` in favour of some name.

Reply:
Actually, that brings up a different point: what I have doesn't work in the first place - what should be happening is that you should be splitting it up into N messages.  The issue indicates that windows does not like to have an iterator past the end of the string(which this code does).  So it really just needs to calculate the N chunks at the beginning and do the correct number of iterations to split up the string properly.